File: docs/legacy(old PACE)/unsupported/VB.NET/PaceWebServiceSDK/SDK.vb
Module SDK

    Public HOST As String = "yourhost"
    Public PORT As Integer = 80
    Public USERNAME As String = "APIUserName"
    Public PASSWORD As String = "APIUserPassword"

    Sub Main()
        While True
            Dim choice As Integer = Prompt()
            If choice = 0 Then
                Console.WriteLine("Exiting...")
                Exit While
            End If
            Select Case choice
                Case 1
                    CreateJob.Run()
                    Exit Select
                Case 2
                    ReadCustomer.Run()
                    Exit Select
                Case 3
                    CloseJob.Run()
                    Exit Select
                Case 4
                    CloneJob.Run()
                    Exit Select
                Case 5
                    FindOpenJobs.Run()
                    Exit Select
                Case 6
                    FindObjects.Run()
                    Exit Select
                Case 7
                    FindObjectsAggregate.Run()
                    Exit Select
                Case 8
                    DeleteObject.Run()
                    Exit Select
                Case 9
                    ConvertToJob.Run()
                    Exit Select
                Case 10
                    InvokeProcessTest.Run()
                    Exit Select
                Case 11
                    ReadGLAccount.Run()
                    Exit Select
                Case 12
                    CreateJobTransactionUsingOverrideStatusAttribute.Run()
                    Exit Select
                Case 13
                    GeoLocateContacts.Run()
                    Exit Select
                Case 14
                    InvokePaceConnect.Run()
                    Exit Select
                Case 15
                    ListKeysForPaceConnectType.Run()
                    Exit Select
                Case 16
                    ListPaceConnectTypes.Run()
                    Exit Select
                Case 17
                    Scheduler.Run()
                    Exit Select
                Case 18
                    AttachmentService.Run()
                    Exit Select
                Case 19
                    GetVersion.Run()
                    Exit Select
                Case 20
                    ReceivePOLine.Run()
                    Exit Select
                Case 21
                    SystemInspector.Run()
                    Exit Select
                Case 22
                    FindObjectDateTimeConstraints.Run()
                    Exit Select
                Case 23
                    ReadJobShipment.Run()
                    Exit Select
                Case 24
                    ItemTemplateSample.Run()
                    Exit Select
                Case 25
                    JobPlanRefreshSample.Run()
                    Exit Select
                Case 26
                    CreateEstimate.Run()
                    Exit Select
                Case 27
                    CreateEstimatePart.Run()
                    Exit Select
                Case 28
                    ReportServiceSample.Run()
                    Exit Select    
                Case 29
                    ' Uncomment the below line for multi company.
                    ' Note : check the documentation for configurations before running this sample.
                    ' CreateJobSample.Run()
                    Exit Select
            End Select
        End While
    End Sub

    Function Prompt() As Integer
        Console.WriteLine("Example programs:")
        Console.WriteLine("1) Create Job")
        Console.WriteLine("2) Read Customer")
        Console.WriteLine("3) Close Job (update)")
        Console.WriteLine("4) Clone Job")
        Console.WriteLine("5) Find Open Jobs")
        Console.WriteLine("6) Find Objects")
        Console.WriteLine("7) Find-objects-aggregate")
        Console.WriteLine("8) Delete Cost Center")
        Console.WriteLine("9) Convert To Job - Invoke Action")
        Console.WriteLine("10) Invoke Process Test - GL Batch Trn Post")
        Console.WriteLine("11) Read GL Account")
        Console.WriteLine("12) Create-JobTransaction-Using-override status attribute")
        Console.WriteLine("13) Geo Locate Contacts")
        Console.WriteLine("14) Invoke Pace Connect")
        Console.WriteLine("15) ListKeys For PaceConnectType")
        Console.WriteLine("16) List Pace Connect Types")
        Console.WriteLine("17) Invoke Scheduler")
        Console.WriteLine("18) Attachment Service")
        Console.WriteLine("19) Get Version")
        Console.WriteLine("20) Receive PO Line")
        Console.WriteLine("21) System Inspector")
        Console.WriteLine("22) FindObjectDateTimeConstraints")
        Console.WriteLine("23) Read job shipment")
        Console.WriteLine("24) Run item template sample")
        Console.WriteLine("25) Refresh Job Plans")        
        Console.WriteLine("26) Create Estimate")
        Console.WriteLine("27) Create Estimate Part")
        Console.WriteLine("28) Execute Report Service Sample")
                        
        Console.WriteLine("29) Run multi company sample for creating job")
        Console.WriteLine("0) Exit")
        Console.Write("Which example would you like to run? ")
        While True
            Try
                Dim choice As Integer = Int32.Parse(Console.ReadLine())
                If (choice < 0) OrElse (choice > 29) Then
                    Continue While
                End If
                Return choice
            Catch
                Console.Write("Invalid choice. Please pick again: ")
            End Try
        End While
        Return -1
    End Function

End Module

File: docs/legacy(old PACE)/C#/4.0/PaceAPISample/SDK.cs
using System;
using Pace_Web_Service_SDK.sample;
using Pace_Web_Service_SDK.util;

namespace Pace_Web_Service_SDK
{
	/// <summary>
	/// Runs the SDK examples.
	/// 
	/// Set the HOST, PORT, USERNAME, and PASSWORD fields for your server.
	/// </summary>
	public class SDK
	{
		public static string HOST = "yourhost";
		public static int PORT = 80;
		public static string USERNAME = "APIUserName";
		public static string PASSWORD = "APIUserPassword";
        public static string PUBLICCOMPANY = "company:public";
        public static string SAMPLECOMPANY = "company:sample";

		public static void Main(string[] args)
		{
            while (true)
            {
                int choice = Prompt();

                if (choice == 0)
                {
                    break;
                }

                switch (choice)
                {
                    case 1:
                        CloseJob.Run();
                        break;
                    case 2:
                        CreateJob.Run();
                        break;
                    case 3:
                        CloneJob.Run();
                        break;
                    case 4:
                        CloneEstimate.Run();
                        break;
                    case 5:
                        FindOpenJobs.Run();
                        break;
                    case 6:
                        InvokeProcessTest.Run();
                        break;
                    case 7:
                        ReadGLAccount.Run();
                        break;
                    case 8:
                        Sample.Run();
                        break;
                    case 9:
                        ReadCustomer.Run();
                        break;
                    case 10:
                        ChargeBackAccountSample.Run();
                        break;
                    case 11:
                        ConvertToJob.Run();
                        break;
                    case 12:
                        CreateJobTransactionUsingOverrideStatusAttribute.Run();
                        break;
                    case 13:
                        FindObjectsAggregate.Run();
                        break;
                    case 14:
                        GeoLocateContacts.Run();
                        break;
                    case 15:
                        FindMoveOnlyJoTransaction.Run();
                        break;
                    case 16:
                        InvokePaceConnect.Run();
                        break;
                    case 17:
                        ListKeysForPaceConnectType.Run();
                        break;
                    case 18:
                        FindObjects.Run();
                        break;
                    case 19:
                        ListPaceConnectTypes.Run();
                        break;
                    case 20:
                        ReadJobShipment.Run();
                        break;
                    case 21:
                        Scheduler.Run();
                        break;
                    case 22:
                        SystemInspector.Run();
                        break;
                    case 23:
                        AttachmentService.Main();
                        break;
                    case 24:
                        Pace_Web_Service_SDK.util.Version.Run();
                        break;
                    case 25:
                        DeleteObject.Run();
                        break;
                    case 26:
                        FindObjectDateTimeConstraints.Run();
                        break;
                    case 27:
                        ReceivePOLine.Run();
                        break;
                    case 28:
                        ItemTemplateSample.Run();
                        break;
                    case 29:
                        JobPlanRefreshSample.Run();
                        break;
                    case 30:
                        CreateEstimate.Run();
                        break;
                    case 31:
                        CreateEstimatePart.Run();
                        break;
                    case 32:
                        ReportServiceSample.Run();
                        break; 
                    case 33:
                        // Uncomment the below line for multi company.
                        // Note : check the documentation for configurations before running this sample.
                        //multicompanysample.CreateJob.Run();
                        break;
                }
            }

			Console.WriteLine("\nHit any key to exit.");
			Console.ReadLine();
		}

        private static int Prompt()
        {
            Console.WriteLine("Example programs:");
            Console.WriteLine("\t1) CloseJob");
            Console.WriteLine("\t2) CreateJob");
            Console.WriteLine("\t3) CloneJob");
            Console.WriteLine("\t4) CloneEstimate");
            Console.WriteLine("\t5) FindOpenJobs");
            Console.WriteLine("\t6) InvokeProcessTest");
            Console.WriteLine("\t7) ReadGLAccount");
            Console.WriteLine("\t8) CRUD Sample");
            Console.WriteLine("\t9) ReadCustomer");
            Console.WriteLine("\t10) Charge-Back-Account-Sample");
            Console.WriteLine("\t11) Convert-To-Job");
            Console.WriteLine("\t12) Create-JobTransaction-Using-override status attribute");
            Console.WriteLine("\t13) Find-objects-aggregate");
            Console.WriteLine("\t14) Geo Locate Contacts");
            Console.WriteLine("\t15) FindMoveOnlyTransaction");
            Console.WriteLine("\t16) Invoke Pace Connect");
            Console.WriteLine("\t17) ListKeys For PaceConnectType");
            Console.WriteLine("\t18) Find Objects");
            Console.WriteLine("\t19) List Pace Connect Types");
            Console.WriteLine("\t20) Read Job Shipment");
            Console.WriteLine("\t21) Invoke Scheduler");
            Console.WriteLine("\t22) System Inspector");
            Console.WriteLine("\t23) Attachment Service");
            Console.WriteLine("\t24) Get Version");
            Console.WriteLine("\t25) Delete Object");
            Console.WriteLine("\t26) FindObjectDateTimeConstraints");
            Console.WriteLine("\t27) Receive PO Line");
            Console.WriteLine("\t28) Run item template sample");
            Console.WriteLine("\t29) Refresh Job Plans");
            Console.WriteLine("\t30) Create Estimate");
            Console.WriteLine("\t31) Add Estimate part");
            Console.WriteLine("\t32) Report service sample");
                                    
            Console.WriteLine("\t0) Exit");
            Console.Write("Which example would you like to run? ");

            while (true)
            {
                try
                {
                    int choice = Int32.Parse(Console.ReadLine());
                    if ((choice < 0) || (choice > 33))
                        continue;
                    return choice;
                }
                catch
                {
                    Console.Write("Invalid choice. Please pick again: ");
                }
            }
        }
    }
}